File: metarealm-growth-os-main/backend/app/schemas/agents.py
from app.schemas.base import CamelModel


class AgentLastRunOut(CamelModel):
    status: str
    summary: str
    created_at: str


class AgentGuideOut(CamelModel):
    does: str
    use: str
    cannot: str


class AgentInfoOut(CamelModel):
    name: str
    title: str
    description: str
    guide: AgentGuideOut | None = None
    last_run: AgentLastRunOut | None = None


class AgentRunRequest(CamelModel):
    topic: str | None = None
    platform: str | None = None
    meeting_id: str | None = None
    company_id: str | None = None
    opportunity_id: str | None = None
    batch: bool | None = None
    reset: bool | None = None
    auto: bool | None = None


class AgentRunOut(CamelModel):
    agent: str
    status: str
    summary: str
    details: dict | None = None

File: metarealm-growth-os-main/frontend/types/agents.ts
/** AI employee domain — mirrors backend/app/schemas/agents.py. */

export type AgentStatus = "completed" | "llm_unavailable" | "error";

export interface AgentRunResult {
  agent: string;
  status: AgentStatus;
  summary: string;
  details?: Record<string, unknown> | null;
}

export interface AgentLastRun {
  status: string;
  summary: string;
  createdAt: string;
}

export interface AgentGuide {
  does: string;
  use: string;
  cannot: string;
}

export interface AgentInfo {
  name: string;
  title: string;
  description: string;
  guide?: AgentGuide | null;
  lastRun?: AgentLastRun | null;
}
